File: Sponsors/mw_streams_python/mws/operators/average.py
from numbers import Number

from mws.operators.operator import Operator


class Average(Operator[Number, Number]):
    def __init__(self) -> None:
        super().__init__()
        self.sum = 0.0
        self.count = 0

    def consume(self, event: Number) -> None:
        raise NotImplementedError("TODO implement me")

File: Sponsors/mw-streams-java/src/main/java/com/mwam/hackathon/operators/Average.java
package com.mwam.hackathon.operators;

public class Average<I extends Number> extends Operator<I, Double> {
    private double sum = 0.0;
    private long count = 0;

    @Override
    public void consume(I input) {
        sum += input.doubleValue();
        count += 1;
        double average = sum / count;

        // the new average to any parent operators
        emit(average);
    }
}
